pallets/traits/src/loans.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021 Parallel Finance Developer.
+// This file is part of Parallel Finance.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use codec::{Decode, Encode};
+use frame_support::dispatch::DispatchError;
+use primitives::{Rate, Ratio};
+use scale_info::TypeInfo;
+use sp_runtime::{FixedU128, RuntimeDebug};
+use sp_std::prelude::*;
+
+pub trait LoansRateProvider<CurrencyId> {
+    fn get_full_interest_rate(asset_id: &CurrencyId) -> Option<Rate>;
+}
+
+pub trait LoansOperator<CurrencyId, AccountId, Balance> {
```

Comment:
I think we can still call it `Loans` because we have pallet_traits::AMM, we can adjust them together in the future if needed

Reply:
first try `Loans` but will not compile seems  some conflict with the name space currently imported